src/wp-includes/functions.php

```diff
@@ -3349,6 +3351,24 @@ function wp_get_image_mime( $file ) {
 		) {
 			$mime = 'image/webp';
 		}
+
+		/**
+		 * Add AVIF fallback detection when image library doesn't support AVIF.
+		 *
+		 * @todo check the third 4-byte token "[major_brand]", for "avif" or "avis" .
```

Comment:
AVIF spec inherits MIAF and HEIF so it relies on ISOBMFF.  
ISOBMFF files start with a `FileTypeBox` whose syntax begins with:
- 4-byte big-endian unsigned integer being the number of bytes used by the `FileTypeBox`, including this field
- 4-byte hard-coded literal string `"ftyp"`
- 4-byte string `"[major_brand]"`
- other fields

So I would check for:
- second 4-byte token being exactly `"ftyp"`
- third 4-byte token being exactly `"avif"` or `"avis"`
- unnecessary but the first three bytes are very likely to be 0 (no more than 60 compatible brands in `FileTypeBox`)

Section `4.3.1 File-type box Definition` of specification ISO/IEC 14496-12 can be cited and https://aomediacodec.github.io/av1-avif/v1.1.0.html#brands can be cited for the major brands that can be reasonably expected in an AVIF file (`avio` should not).

Reply:
Adopted this approach in 3ecb6825eeb5a8b224a115a07a19906fd457270a

Reply:
This worked well in my testing and simplified the logic here; thanks for the tip!